src/aleph/netty.clj

```diff
@@ -1400,7 +1412,10 @@ initialize an DnsAddressResolverGroup instance.
                 "ssl-handler"
                 (let [ch (.channel p)]
                   (if remote-address
-                    (ssl-handler ch ssl-ctx remote-address)
+                    ;; Always passing `nil` as `ssl-endpoint-id-alg` here since this code path is
+                    ;; only hit in the deprecated `create-client`. To keep it fully backwards
+                    ;; compatible, we don't enable endpoint identification by default.
```

Comment:
Hm I wonder now whether this is really the wisest thing to do as it leaves users of the deprecated API vulnerable. Why not make it a (mildly) breaking change for them as well ... I think I'll go ahead and change this.

Reply:
OK pushed a commit to that effect.